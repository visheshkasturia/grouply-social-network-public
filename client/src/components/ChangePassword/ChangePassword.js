/*eslint-disable */
import { useState } from "react";
import { setChangePassword } from "../../apiCalls/setChangePassword";

export default function ChangePassword( {user} ) {
  const [inputs, setInputs] = useState({});
  const [status, setStatus] = useState();

  const handleChange = (event) => {
    const name = event.target.name;
    const value = event.target.value;
    setInputs(values => ({...values, [name]: value}))
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    inputs.createdBy = user._id;
    inputs.email = user.email;
    setChangePassword(inputs).then((response) => {
      setStatus(response);
    });
  }

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label>Enter Current Password:
        <input 
          type="password" 
          name="current" 
          value={inputs.current || ""} 
          onChange={handleChange}
        />
        </label>
        <label>Enter New Password:
        <input 
          type="password" 
          name="new" 
          value={inputs.new || ""} 
          onChange={handleChange}
        />
        </label>
          <button type="submit"> Change Password </button>
      </form>
      <div> {status} </div>
    </div>
  )
}