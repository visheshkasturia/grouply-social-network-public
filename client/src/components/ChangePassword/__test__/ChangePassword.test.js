import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ChangePassword from '../ChangePassword'

it('ChangePassword renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChangePassword />, div);
});

it('ChangePassword matches snapshot', () => {
  const tree = renderer.create(<ChangePassword />).toJSON();
  expect(tree).toMatchSnapshot();
});