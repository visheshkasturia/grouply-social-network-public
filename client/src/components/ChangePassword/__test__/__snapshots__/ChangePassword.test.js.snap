// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Enter Current Password:
      <input
        name="current"
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
    <label>
      Enter New Password:
      <input
        name="new"
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
       Change Password 
    </button>
  </form>
  <div>
     
     
  </div>
</div>
`;
